### PR TITLE
Fixes #17549: add /pulp to list to not remove trailing slash.

### PR DESCRIPTION
**Pull request by waldenraines**
Removing the trailing slash from Pulp URLs breaks them so this change
adds an ignore list so we don't remove the trailing slash from these
URLs.
    
http://projects.theforeman.org/issues/17549

### CODE REVIEW

**Comment by waldenraines** [comment]
[test]

**Comment by ehelms** [comment]
`/gpg_keys/1/` now results in a 404 instead of taking me to `/gpg_keys/1` which seems wrong to me because the error is non-obvious that its a trailing slash issue. Thoughts?

**Comment by waldenraines** [comment]
> /gpg_keys/1/ now results in a 404 instead of taking me to /gpg_keys/1 which seems wrong to me because the error is non-obvious that its a trailing slash issue. Thoughts?

Yeah, that was the trade off here to get the pulp urls working.  It is technically a 404 and I don't know of a good way that's not hackish to get redirection on slashes unless the url contains "pulp".

**Comment by ehelms** [comment]
Wouldn't we want to just ignore processing any links that start with `/pulp` and allow them to do whatever it is they would normally do?

**Comment by waldenraines** [comment]
> Wouldn't we want to just ignore processing any links that start with /pulp and allow them to do whatever it is they would normally do?

We could do that instead.  I just didn't want to have to manually exclude `pulp/` but I can add an exclude list.

**Comment by ehelms** [comment]
I think that would be cleaner in the end.

**Comment by waldenraines** [comment]
> I think that would be cleaner in the end.
> 
> If we didn't own the root URL space this wouldn't be as much an issue.

@ehelms updated.

File: test/routing.module.test.js [hunk 1]
```diff
@@ -35,6 +35,11 @@ describe('config: Bastion.routing', function () {
             goTo('/state///');
             expect($window.location.href).toBe('http://server/state');
         });
+
+        it("doesn't remove slashes if whitelisted", function () {
+            goTo('/pulp/');
```
Comment (ehelms) [inline test/routing.module.test.js]: Mind testing `/pulp/repos/` since that is a more typical path to go to?

Comment (waldenraines) [inline test/routing.module.test.js]: Done.

**Comment by waldenraines** [comment]
@ehelms updated.

**Comment by waldenraines** [comment]
[test]

**Pull request merged**